### DISCUSSION TITLE
How to define `select`from `useQuery` asynchronous?

### DISCUSSION THREAD

**Discussion by frangeris** [TanStack/query#8175] [Q&A]
I'm making some experiments where the `data` is asynchronous, so, how can I define the `select` method `async?`

Here's the use case I'm trying to implement:

```ts
  const { data } = useQuery({
    queryKey: ["tests"],
    queryFn: () => getToApi(),
    select: (d) => await d.data, // <--- this
  });
```
`select` property can't be async due to data will return a promise

Is it viable to add support for something like a new `asyncSelect` method?

Without thinking in handling the `data promise` on a different function or hook, can it be done using `useQuery` definition or global `select` query options?

_ps: this is a similar way of getting this, but with `queryFn`_

```ts
//...
queryFn: async () => (await getToApi()).data,
```

Thanks a lot! 🙌🏽

**Comment by TkDodo** [comment]
select isn’t asynchronous. If you want async transformation, you have to do it in the queryFn.
